### PR TITLE
IDEM-45

### PR DESCRIPTION
**Pull request by siarangoh**
- Add functionality to select from device button
- Allow the user to see the images/files when clicking on them

https://user-images.githubusercontent.com/24626745/171805841-fc87ab09-31ee-42b0-8f80-f00c1ec3d6e0.mp4

IDEM-45 (1)

IDEM-45 (2)

### CODE REVIEW

**Review by jrocca82 — DISMISSED** [review]

**Review by JasminFarhi — APPROVED** [review]

File: app/src/navigation/screens/DocumentsScreen.tsx [hunk 1]
```diff
@@ -46,6 +46,19 @@ const DocumentsScreen: React.FC = () => {
...
+  const uploadFile = async () => {
+    try {
+      const res = await DocumentPicker.getDocumentAsync({
+        type: "*/*"
+      });
+      if (res && res.type !== "cancel") {
+        addFile(selectedDocumentId, res);
+      }
+    } catch (error) {
+      //
```
Comment (xiangzhe0812) [inline app/src/navigation/screens/DocumentsScreen.tsx:58]: Need do something if catched the error

File: app/src/context/DocumentStore.tsx [hunk 2]
```diff
@@ -39,23 +41,27 @@ export const DocumentProvider: React.FC<{
 
   const addFile = async (
     documentId: DocumentId,
-    file: ImagePicker.ImageInfo
+    file: ImagePicker.ImageInfo | DocumentPicker.DocumentResult
```
Comment (xiangzhe0812) [inline app/src/context/DocumentStore.tsx:44]: Only need to pass in uri.
@jrocca82

File: app/src/context/DocumentStore.tsx [hunk 3]
```diff
@@ -39,23 +41,27 @@ export const DocumentProvider: React.FC<{
...
   ): Promise<string> => {
+    if (file.type === "cancel") {
```
Comment (xiangzhe0812) [inline app/src/context/DocumentStore.tsx:46]: If pass in uri then no need this if statement

**Review by bitcoinbrisbane — APPROVED** [review]
LGTM

File: app/src/navigation/screens/DocumentsScreen.tsx (hunk 1 above)
Comment (bitcoinbrisbane) [reply app/src/navigation/screens/DocumentsScreen.tsx]: @jrocca82 this is now resolved?

Comment (jrocca82) [reply app/src/navigation/screens/DocumentsScreen.tsx]: Yes, does console.log(error)

**Review by xiangzhe0812 — APPROVED** [review]

**Review by bitcoinbrisbane — DISMISSED** [review]

**Review by jrocca82 — APPROVED** [review]

**Review by xiangzhe0812 — APPROVED** [review]

**Pull request merged**